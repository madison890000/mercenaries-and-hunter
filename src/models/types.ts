import StringWithID from './StringWithID';

namespace DataModel {
    export enum Degree {
        Bachelor = 'BACHELOR',
        Master = 'MASTER',
        PhD = 'PHD'
    }

    export type Gender = 'MALE' | 'FEMALE';

    export type IAchievement = {
        text: string;
        categories: string[];
    };
    export type Achievement = {
        id: string;
        text: string;
        categories: StringWithID[];
    };
    export type SolutionsOfHowToImplement = string;

    export enum JobPositionLevel {
        Staff = 'STAFF',
        Senior = 'SENIOR',
        Middle = '',
        Junior = 'JUNIOR'
    }

    export type JobPosition = 'Software Engineer' | 'Front-End Engineer' | 'Back-End Engineer' | 'Full-Stack Engineer';

    export type Job = {
        position: JobPosition;
        level: JobPositionLevel;
    };
    export enum SkillLevel {
        understand = 'UNDERSTAND',
        familiar = 'FAMILIAR',
        proficient = 'PROFICIENT'
    }

    export enum Importance {
        Essential,
        Advanced
    }

    export enum CompanyType {
        Enterprise,
        Startup
    }
}
export default DataModel;
