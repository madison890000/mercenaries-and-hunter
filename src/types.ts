export enum SendType {
    SEND = 0,
    START_INTERVIEWING = 1,
    MIDDLE_INTERVIEWING = 2,
    FINAL_INTERVIEWING = 3,
    OFFER = 4,
    REJECTED = 5,
}

export interface IAppliedInExtension {
    site: string;
    time: string;
    id: string;
    title: string;
    originUrl?: string
}

export interface IAppliedInLocal extends IAppliedInExtension {
    status: SendType;
    like: number;
    needSync: boolean;
    serverId?: number;
    updateTime?: number;
}

export interface IAppliedInServer extends Omit<IAppliedInLocal, 'id'> {
    status: SendType;
    id: number;
    like: number;
    needSync: boolean;
    serverId: number;
    updateTime: number;
}

export type NeedSyncAppliedItem = Omit<IAppliedInLocal, 'id'> & { id?: number }