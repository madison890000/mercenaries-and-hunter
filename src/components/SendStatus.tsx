import Select from '@mui/material/Select';
import MenuItem from "@mui/material/MenuItem";
import {defineMessages, useIntl} from "react-intl";

const messages = defineMessages({
    send: {
        id: 'send-enum.send',
        defaultMessage: "Reload Data",
    },
    startInterview: {
        id: 'send-enum.startInterview',
    },
    middleInterview: {
        id: 'send-enum.middleInterview',
    },
    finalInterview: {
        id: 'send-enum.finalInterview',
    },
    offer: {
        id: 'send-enum.offer',
    },
    rejected: {
        id: 'send-enum.rejected',
    },
});

interface SendStatusProps {
    value: string | number | undefined;
    onChange: any;
}

enum SendType {
    SEND = 0,
    START_INTERVIEWING = 1,
    MIDDLE_INTERVIEWING = 2,
    FINAL_INTERVIEWING = 3,
    OFFER = 4,
    REJECTED = 5,
}

const colors = {
    [SendType.SEND]: 'black',
    [SendType.START_INTERVIEWING]: '#000099',
    [SendType.MIDDLE_INTERVIEWING]: 'var(--color-blue-6)',
    [SendType.FINAL_INTERVIEWING]: '#ff33cc',
    [SendType.OFFER]: 'var(--color-primary)',
    [SendType.REJECTED]: 'var(--color-red)',
}
const SendStatus: React.FC<SendStatusProps> = ({value, onChange}) => {

    const intl = useIntl();
    return (
        <div style={{
            margin: 10
        }}>
            <Select
                value={value}
                style={{
                    minWidth: 100,
                    // @ts-ignore
                    color: colors[value],
                }}
                onChange={onChange}
                variant="standard"
            >
                <MenuItem value={SendType.SEND}>{intl.formatMessage(messages.send)}</MenuItem>
                <MenuItem value={SendType.START_INTERVIEWING}
                          style={{
                              color: '#000099'
                          }}
                >{intl.formatMessage(messages.startInterview)}</MenuItem>
                <MenuItem value={SendType.MIDDLE_INTERVIEWING}
                          style={{
                              color: 'var(--color-blue-6)'
                          }}
                >{intl.formatMessage(messages.middleInterview)}</MenuItem>
                <MenuItem value={SendType.FINAL_INTERVIEWING}
                          style={{
                              color: '#ff33cc',
                          }}
                >{intl.formatMessage(messages.finalInterview)}</MenuItem>
                <MenuItem value={SendType.OFFER}
                          style={{
                              color: 'var(--color-primary)'
                          }}
                >{intl.formatMessage(messages.offer)}</MenuItem>
                <MenuItem value={SendType.REJECTED} style={{
                    color: 'var(--color-red)'
                }}>{intl.formatMessage(messages.rejected)}</MenuItem>
            </Select>
        </div>
    )
}

export default SendStatus