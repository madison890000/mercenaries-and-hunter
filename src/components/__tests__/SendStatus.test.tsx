import React from 'react';
import SendStatus from '../SendStatus';
import renderer from 'react-test-renderer';
import {SendType} from "../../types";

test('SendStatus -> Send', () => {
    const DividerDom = renderer.create(
        <SendStatus value={SendType.SEND} />
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('SendStatus -> OFFER', () => {
    const DividerDom = renderer.create(
        <SendStatus value={SendType.OFFER} />
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('SendStatus -> FINAL_INTERVIEWING', () => {
    const DividerDom = renderer.create(
        <SendStatus value={SendType.FINAL_INTERVIEWING} />
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('SendStatus -> REJECTED', () => {
    const DividerDom = renderer.create(
        <SendStatus value={SendType.REJECTED} />
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('SendStatus -> MIDDLE_INTERVIEWING', () => {
    const DividerDom = renderer.create(
        <SendStatus value={SendType.MIDDLE_INTERVIEWING} />
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('SendStatus -> START_INTERVIEWING', () => {
    const DividerDom = renderer.create(
        <SendStatus value={SendType.START_INTERVIEWING} />
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});
