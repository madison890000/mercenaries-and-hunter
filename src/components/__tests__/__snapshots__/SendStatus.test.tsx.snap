// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendStatus -> FINAL_INTERVIEWING 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    onClick={[Function]}
    style={
      Object {
        "color": "#ff33cc",
        "minWidth": 100,
      }
    }
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Final Interview
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value={3}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SendStatus -> MIDDLE_INTERVIEWING 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--color-blue-6)",
        "minWidth": 100,
      }
    }
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Middle Interview
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value={2}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SendStatus -> OFFER 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--color-primary)",
        "minWidth": 100,
      }
    }
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Offer
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value={4}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SendStatus -> REJECTED 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--color-red)",
        "minWidth": 100,
      }
    }
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Rejected
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value={5}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SendStatus -> START_INTERVIEWING 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    onClick={[Function]}
    style={
      Object {
        "color": "#000099",
        "minWidth": 100,
      }
    }
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Start Interview
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value={1}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SendStatus -> Send 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary  css-74e2me-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
        "minWidth": 100,
      }
    }
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Applied
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value={0}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
